lib/view.js

```diff
@@ -37,7 +38,19 @@ export default class AdvancedOpenFileView {
 
         // Update the path list whenever the path changes.
         this.pathEditor.onDidChange(() => {
-            let newPath = new Path(this.pathEditor.getText());
+            let path = this.pathEditor.getText();
+
+            if (config.get("backspaceThroughDir")) {
+                if (path + "/" === this.prevPath) {
+                    path = path.replace(/[^/]+$/, "");
+                    this.prevPath = path;
+                    this.pathEditor.setText(path)
+                    // let setText trigger new onDidChange
+                    return
+                }
+            }
+            this.prevPath = path;
+            let newPath = new Path(path);
```

Comment:
We should move this logic to [`AdvancedOpenFileController.pathChange`](https://github.com/Osmose/advanced-open-file/blob/master/lib/controller.js#L66). That's the main spot for making data changes in response to the path changing. The view is only interested in updating the UI in response to the data changing.

Reply:
Sure, seems like this 5-minute hack will graduate from kindergarten then :)